src/Kestrel.Core/Internal/HttpConnection.cs

```diff
@@ -303,7 +305,8 @@ private void DisposeAdaptedConnections()
         private HttpProtocols SelectProtocol()
         {
             var hasTls = _context.ConnectionFeatures.Get<ITlsConnectionFeature>() != null;
-            var applicationProtocol = _context.ConnectionFeatures.Get<ITlsApplicationProtocolFeature>()?.ApplicationProtocol;
+            var applicationProtocol = _context.ConnectionFeatures.Get<ITlsApplicationProtocolFeature>()?.ApplicationProtocol
```

Comment:
Can't we now just get rid of ITlsApplicationProtocolFeature and pull the application protocal from the ITlsConnectionFeature we access a line above?

Reply:
ITlsConnectionFeature is from HttpAbstractions, it's a breaking change to merge them.

Reply:
Oh, I see what you mean. The implementation is already merged but we're accessing it via interfaces and can't tell.

Reply:
Just delete ITlsApplicationProtocolFeature.cs

Reply:
I see. ITlsConnectionFeature itself doesn't include ApplicationProtocol though the implementation does. Just get rid of the TODO then.